fennel/datasets/test_dataset.py

```diff
@@ -2930,3 +2930,154 @@ def pipeline_window(cls, app_event: Dataset):
     assert operator_req == expected_operator_request, error_message(
         operator_req, expected_operator_request
     )
+
+
+def test_erase_key():
+    @meta(owner="nitin@fennel.ai")
+    @dataset
+    class PageViewEvent:
+        user_id: str
+        page_id: str
+        t: datetime
+
+    @meta(owner="nitin@fennel.ai")
+    @dataset
+    class Sessions:
+        user_id: str = field(key=True, erase_key=True)
+        avg_session_secs: float
+        t: datetime
+
+        @pipeline(version=1)
+        @inputs(PageViewEvent)
+        def pipeline_window(cls, app_event: Dataset):
+            sessions = (
```

Comment:
nit: since the test is asserting for erase_key, most of the asserts (what is the pipeline, operators in it, etc). Also we can cut short the pipeline itself to keep the test case succinct